.template-helpers/variant.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -e
+
+repo="$1"
+if [ -z "$repo" ]; then
+	echo >&2 "usage: $0 repo"
+	echo >&2 "   ie: $0 hylang"
+	exit 1
+fi
+
+dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+IFS=$'\n'
+lines=( $(curl -sSL 'https://raw.githubusercontent.com/docker-library/official-images/master/library/'"$repo" | grep -vE '^$|^#') )
+unset IFS
+
+tags=()
+# this loop finds all the "variants", making assumptions about what they can contain
+for line in "${lines[@]}"; do
+	tag="$(echo "$line" | awk -F ': +' '{ print $1 }' | sed -r 's/[0-9.-]+//g')"
```

Comment:
Since `variant-onbuild.md` assumes `%%REPO%%:onbuild` is a valid tag, and since all current repos have that property, maybe we should just simplify this to grabbing all the tags of the repo as-is without the `sed`?  It should be pretty harmless to be checking for, for example, `variant-3.4-onbuild.md` to exist, since we'll eventually get down to just `variant-onbuild.md` which does exist.  Thoughts?